auth_backends/tests/test_pipeline.py

```diff
@@ -40,20 +40,67 @@ class UpdateEmailPipelineTests(TestCase):
 
     def setUp(self):
         super().setUp()
-        self.user = User.objects.create()
+        self.user = User.objects.create(username='test_user')
+        self.strategy = MagicMock()
 
-    def test_update_email(self):
-        """ Verify that user email is updated upon changing email. """
+        # Make strategy.storage.user.changed actually save the user
+        def save_user(user):
+            user.save()
+
+        self.strategy.storage.user.changed.side_effect = save_user
+
+    @patch('auth_backends.pipeline.set_custom_attribute')
+    def test_update_email(self, mock_set_attribute):
+        """ Verify that user email is updated upon changing email when usernames match. """
         updated_email = 'updated@example.com'
         self.assertNotEqual(self.user.email, updated_email)
 
-        update_email(load_strategy(), {'email': updated_email}, user=self.user)
+        # Provide matching username in details
+        update_email(self.strategy, {'email': updated_email, 'username': 'test_user'}, user=self.user)
+
+        # Verify email was updated
         self.user = User.objects.get(username=self.user.username)
         self.assertEqual(self.user.email, updated_email)
+        self.strategy.storage.user.changed.assert_called_once_with(self.user)
```

Comment:
This assertion on a mock seems unnecessary, since you've already asserted that the email was updated. Can you explain why this is needed?